evaluator.py

```diff
@@ -0,0 +1,186 @@
+import argparse
+import os
+import sys
+import yaml
+import json
+from collections import namedtuple
+from common import TestCaseType, TestType
+from logger import logger
+
+
+Bitrate = namedtuple("Bitrate", "tx rx")
+
+#TODO: We made need to extend this to include results from other plugins (i.e. is HWOL working) such that
+# we can return a single "Status" from a test
+Status = namedtuple("Status", "result num_passed num_failed")
+
+class Result():
+    def __init__(self, test_id: TestCaseType, test_type: TestType, success: bool, bitrate_gbps: Bitrate):
+        self.test_id = test_id
+        self.test_type = test_type
+        self.success = success
+        self.bitrate_gbps = bitrate_gbps
+
+    def dump_to_json(self) -> dict:
+        return {"test_id": self.test_id,
```

Comment:
I would arguing python dictionaries are easier to work with since converting a dict to json requires 1 less command than converting a dataclass. I believe with the dataclass method you would first have to convert the dataclass to a dict i.e. ```json.dumps(dataclasses.asdict(x))```  versus ```json.dumps(my_dict)```.

Reply:
That's true, however we end up losing the nice schematization that a dataclass provides. Anyone can change the dict at any time and we do not have type safety. 

It's not the end of the world if it stays as a dict, but if we want this to be a machine readable format, we probably want to make sure the format is defined somewhere.

Reply:
That's a really good point. So perhaps a dataclass would be best to ensure the internal representation of the data remains consistent